src/parser/parser.rs

```diff
@@ -558,20 +558,18 @@ impl<'cmd> Parser<'cmd> {
             if self.cmd.is_infer_subcommands_set() {
                 // For subcommand `test`, we accepts it's prefix: `t`, `te`,
                 // `tes` and `test`.
-                let v = self
-                    .cmd
-                    .get_subcommands()
-                    .filter_map(|s| {
-                        if s.get_name().starts_with(arg) {
-                            return Some(s.get_name());
-                        }
+                let mut iter = self.cmd.get_subcommands().filter_map(|s| {
+                    if s.get_name().starts_with(arg) {
+                        return Some(s.get_name());
+                    }
 
-                        s.get_all_aliases().find(|s| s.starts_with(arg))
-                    })
-                    .collect::<Vec<_>>();
+                    s.get_all_aliases().find(|s| s.starts_with(arg))
+                });
 
-                if v.len() == 1 {
-                    return Some(v[0]);
+                if let name @ Some(_) = iter.next() {
+                    if iter.next().is_none() {
```

Comment:
As this feature is more in the periphery, I care more about binary size than performance.